example_sns_init.yaml

```diff
@@ -25,96 +31,338 @@ description: >
+# of the SNS.
 Token:
+    # The name of the token issued by the SNS ledger.
+    # Must be a string with a length between 4 and 255 characters.
+    #
+    # Example: InternetComputerProtocol
     name: Rock Out Token
+
+    # The symbol of the token issued by the SNS Ledger.
+    # Must be a string with a length between 3 and 10 characters.
+    #
+    # Example: ICP
     symbol: ROT
-    transaction_fee: 10_000 e8s # You can leave this alone.
 
-# It is fine to leave these alone.
+    # SNS ledger transactions will have this fee.
+    transaction_fee: 10_000 e8s
+
+    # Path to the SNS token logo on your local filesystem. The path is relative
+    # to the configuration file location, unless an absolute path is given.
+    # Must have less than 341,334 characters, roughly 256 Kb. The only supported
+    # format is PNG.
+    logo: logo.png
+
+# Configures SNS proposal-related fields. These fields have no
+# effect on the NNS proposal that is created from this file, but will affect all
+# SNS proposals submitted to the newly created SNS.
 Proposals:
+    # The cost of making an SNS proposal that is rejected by the SNS Neuron
+    # holders.
     rejection_fee: 1 token
+
+    # The initial voting period of a newly-created SNS proposal. A proposal's
+    # voting period may be increased during its lifecycle due to the
+    # wait-for-quiet algorithm (see details below).
     initial_voting_period: 4d
+
+    # The wait for quiet algorithm extends the voting period of a proposal when
+    # there is a flip in the majority vote during the proposal's voting period.
+    #
+    # Without this, there could be an incentive to vote right at the end of a
+    # proposal's voting period, in order to reduce the chance that people will
+    # see and have time to react to that.
+    #
+    # If this value is set to 1 day, then a change in the majority vote at the
+    # end of a proposal's original voting period results in an extension of the
+    # voting period by an additional day. Another change at the end of the
+    # extended period will cause the voting period to be extended by another
+    # half-day, etc.
+    #
+    # The total extension to the voting period will never be more than twice
+    # this value.
+    #
+    # For more information, please refer to
+    # https://wiki.internetcomputer.org/wiki/Network_Nervous_System#Proposal_decision_and_wait-for-quiet
     maximum_wait_for_quiet_deadline_extension: 1 day
 
-# It is fine to leave these alone.
+# Configuration of SNS neuron-related fields. These fields have no effect on the
+# NNS proposal, but will affect the SNS itself.
 Neurons:
-    minimum_creation_stake: 10 tokens
+    # The minimum amount of SNS tokens an SNS ledger account must have to stake
+    # a neuron. The larger the minimum, the more tokens must be acquired to be
+    # able to vote.
+    minimum_creation_stake: 1 tokens
 
-# It is fine to leave these alone.
```

Comment:
We removed some of these comments. IIRC there are some values that must not be filled out and will just be filled in with defaults if left empty. Is this still the case? If so, should this file somehow say which ones are mandatory and which one are optional?

Reply:
AFAIK, we're currently precise about 
1. which fields are optional (e.g., `confirmation_text`, `restricted_countries`)
2. which fields must not be set (`Principals`)
3. which fields must be set in for a mainnet deployment but will be set automatically in testing (see `YOUR_PRINCIPAL_ID` and `YOUR_CANISTER_ID`

Am I missing a category?

Reply:
Two follow up questions:
1. where are we precise about that / where do we say that?
2. IIRC it used to be the case that one didn't have to give a min_stake or a max_dissolve_delay and if one didn't, then the tool would just take the default values (those from the NNS in most cases). If this is not the case anymore, then this is probably fine - I just wanted to make sure